impl/src/main/java/io/kroxylicious/testing/kafka/common/SaslPlainAuth.java

```diff
@@ -32,6 +32,9 @@
     /**
      * The interface User password.
      */
+    @Target({ ElementType.FIELD, ElementType.PARAMETER })
+    @Retention(RetentionPolicy.RUNTIME)
+    @KafkaClusterConstraint
```

Comment:
I haven't experience of meta annotations, but `@ @KafkaClusterConstraint` looks wrong.  I think we want to express a constraint that says this annotation may only appear within the @SaslPlainAuth annotation.  So I think you want to introduce another kind of constraint check.

Reply:
I agree, I'm a bit lost here, but If I don't set this constraint I cannot use this kind of annotation as requested:
```
@SaslPlainAuth.UserPassword(user="XXX", password="XXX") KafkaCluster cluster
```
In the `KafkaClusterConfig` class it never creates the map for the `users` because it is not recognize as a proper annotation for the `KafkaCluster`.

Reply:
did you try?

```java
@SaslPlainAuth{users={@UserPassword(user="XXX", password="XXX")}) KafkaCluster cluster
```